### DISCUSSION TITLE
authstate routing in GetMaterialApp

### DISCUSSION THREAD

**Discussion by toklive** [jonataslaw/getx#1346] [Q&A]
Hello Team - 

Are there any example of how to route to different pages based on auth state..

In my case , 

User open the app for first time --> Intro --> Login --> Home page

I am trying to see how I can implement something below in GetMatielApp()

import 'package:firebase_auth/firebase_auth.dart';
import 'package:flutter/material.dart';
import 'package:phone_auth_example/homePage.dart';
import 'signUpPage.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
        debugShowCheckedModeBanner: false,
        home: StreamBuilder(
          stream: FirebaseAuth.instance.onAuthStateChanged,
          builder: (ctx, userSnapshot) {
            if (userSnapshot.hasData) {
              return HomePage();
            } else if (userSnapshot.hasError) {
              return CircularProgressIndicator();
            }
            return LoginPage();
          },
        ));
  }
}

I am using Get CLI and the default is like below..I need to control the routing based on authentication state within the getPages...      getPages: AppPages.routes,

 Widget build(BuildContext context) {
    return GetMaterialApp(
      title: "MyApp",
      initialRoute: AppPages.INITIAL,
      getPages: AppPages.routes,
      theme: lightThemeData(context),
    );
  }

**Comment by RowanG1** [comment]
Hi, did you find any solution to this?

**Comment by RowanG1** [comment]
Solution found:
   ```
  Widget build(BuildContext context) {

   final loginController = Get.find<LoginController>();

// Use listener here. I'm not sure if garbage collection is handled automatically at time of writing.
   loginController.isLoggedIn.listen((val) {
      val ? Get.toNamed("/home") : Get.toNamed("/signin");
    });

    return GetMaterialApp(...);
```
